cache/common.go

```diff
@@ -2,6 +2,44 @@ package cache
 
 import (
 	"errors"
+	"fmt"
+
+	"github.com/buildpack/imgutil/remote"
+
+	"github.com/buildpack/lifecycle"
+	"github.com/buildpack/lifecycle/auth"
+	"github.com/buildpack/lifecycle/cmd"
 )
 
 var errCacheCommitted = errors.New("cache cannot be modified after commit")
+
+// MaybeCache returns a Cache object if one can be created from given arguments or nil.
+func MaybeCache(image, dir string) (lifecycle.Cache, error) {
```

Comment:
I understand the utility of this function but it feels weird to me for it to live here, from a package interface perspective. Given that the logic here is more about the user interface of the cli (which flag or env var has precedence) than the implementation of the cache, I would suggest moving it to the `cmd` package.

Reply:
I could see putting just the contents of the `if` block in a function in the `cache` package, to reduce biolerplate construction logic required for a remote `ImageCache`.